tools/maint/run_python_compiler.bat

```diff
@@ -8,6 +8,8 @@
 :: N.b. In Windows .bat scripts, the ':' character cannot appear inside any if () blocks,
 :: or there will be a parsing error.
 
+@set EM_LAUNCHER=1
```

Comment:
This won't work the way that was intended.

This will set the environment variable permanently to the calling terminal. So then the terminal will be polluted with this environment variable.

I see there is a line `del os.environ['EM_LAUNCHER']` added below, but that does not remove the environment variable for the calling terminal.

So e.g.
```
emcc hello_world.c -o a.js
/path/to/my/python emcc.py hello_world.c -o a.js
```
will go undetected.

Reply:
So I guess we need to under EM_LAUNCHER as soon as launched process is done?   What is the best way to do that?

Reply:
Using a pattern like

```
set EM_LAUNCHER=1
python emcc.py ...
set EM_LAUNCHER=
```
has the problem that it will be unsafe for pressing ctrl-c to abort in the middle of compilation, as the last line would not get executed, and the env. var. would leak again.

What was the rationale for locating `set EM_LAUNCHER=1` outside `setlocal`? If it was after `setlocal`, then it would not leak outside.